Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs

```diff
@@ -0,0 +1,210 @@
+        [SerializeField] private string _volumeName;
+        [SerializeField] private TMP_Text _titleText;
+        [SerializeField] private VolumeType _volumeType = VolumeType.VCA;
+        
+        private Slider _customSlider = null;
+        private TMP_InputField _inputField = null;
+
+        private VolumeWrapper _volume;
+
+        private Slider CustomSlider
+        {
+            get
+            {
+                if (_customSlider == null)
+                {
+                    _customSlider = GetComponentInChildren<Slider>();
+                }
+
+                return _customSlider;
+            }
+        }
+
+        private TMP_InputField CustomInputField
+        {
+            get
+            {
+                if (_inputField == null)
+                {
+                    _inputField = GetComponentInChildren<TMP_InputField>();
+                }
+
+                return _inputField;
+            }
+        }
+
+        private float SliderValue
+        {
+            get
+            {
+                return CustomSlider.value / REMAP_VALUE;
+            }
+            set
+            {
+                CustomSlider.value = value * REMAP_VALUE;
+                SetVolume(value);
+            }
+        }
+
+        private void Awake()
+        {
+            CustomSlider.onValueChanged.AddListener(HandleOnSliderValueChanged);
+            CustomInputField.onEndEdit.AddListener(HandleOnInputFieldEnded);
+
+            if (_titleText != null)
+            {
+                _titleText.text = _volumeName;
+            }
+
+            switch (_volumeType)
+            {
+                case VolumeType.VCA:
+                    _volume = new VCAVolume(_volumeName);
+                    break;
+                case VolumeType.Bus:
+                    _volume = new BusVolume(_volumeName);
+                    break;
+                case VolumeType.MasterBus:
+                    _volume = new MasterVolume();
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+            SliderValue = GetVolume();
+
+        }
+
+        private float GetVolume()
+        {
+            if (_volume.GetVolume(out float volume))
+            {
+                return volume;
+            }
+            return -1;
+        }
+
+        private void SetVolume(float volume)
+        {
+            _volume.SetVolume(volume);
+        }
+
+        private void OnDestroy()
+        {
+            CustomSlider.onValueChanged.RemoveListener(HandleOnSliderValueChanged);
+            CustomInputField.onEndEdit.RemoveListener(HandleOnInputFieldEnded);
+        }
+
+        private void HandleOnSliderValueChanged(float value)
+        {
+            CustomInputField.text = value.ToString("F0");
+            SetVolume(SliderValue);
+        }
+
+        private void HandleOnInputFieldEnded(string value)
+        {
+            if (String.IsNullOrWhiteSpace(value))
```

Comment:
should this be `string.IsNullOrEmpty(volume)` ?

Reply:
This included a bunch of WIP code from my changes, separate to what Will was working on. Just reverted my commit to clean up the PR